universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java

```diff
@@ -120,30 +135,42 @@ private ContainerTag buildNewQuestionForm(QuestionForm questionForm) {
   }
 
   private ContainerTag buildEditQuestionForm(long id, QuestionForm questionForm) {
-    ContainerTag formTag = buildQuestionForm(questionForm);
+    ContainerTag formTag = buildQuestionForm(questionForm, true);
     formTag
         .withAction(controllers.admin.routes.QuestionController.update(id).url())
         .with(submitButton("Update").withClass(Styles.ML_2));
     return formTag;
   }
 
-  private ContainerTag buildQuestionForm(QuestionForm questionForm) {
+  private ContainerTag buildQuestionForm(QuestionForm questionForm, boolean submittable) {
```

Comment:
Nit (non-blocking): prefer having a readOnly version of the form building function to reusing the same function for two purposes.

Reply:
Are you sure?  That's pretty verbose and duplicates a lot of logic that will be error-prone - people will change logic in one place but not the other.

Reply:
As long as the two code path only have one reason to change, i.e. they will always change together, then this is all fine. I am just concerned if that is not the case.

If my concern is moot, please add two wrapper functions so at the call site it is more readable.

```
private ContainerTag buildReadOnlyQuestionForm(QuestionForm questionForm) {
  buildQuestionForm(questionForm, false);
}

private ContainerTag buildEditableQuestionForm(QuestionForm questionForm) {
  buildQuestionForm(questionForm, true);
}
```

Reply:
Done.